hnix-store-core/src/System/Nix/ReadonlyStore.hs

```diff
@@ -28,5 +28,14 @@ makeTextPath nm h refs = makeStorePath ty h nm
   where
     ty = BS.intercalate ":" ("text" : map storePathToRawFilePath (HS.toList refs))
 
+makeFixedOutputPath :: (KnownStoreDir storeDir, ValidAlgo hashAlgo, NamedAlgo hashAlgo) => Bool -> Digest hashAlgo -> StorePathName -> StorePath storeDir
+makeFixedOutputPath recursive h nm =
+  makeStorePath ty h' nm
+  where
+    (ty, h') =
+      if recursive
```

Comment:
Only if recursive and sha256, right?

Reply:
https://github.com/NixOS/nix/blob/41ba5135e0a2d9940c4e8cbedff44eb1a47af782/src/libstore/store-api.cc#L200

Reply:
Ah maybe that's why I had it fixed to SHA256 previously. Will update!

Reply:
@shlevy I've updated, it will now only use "source" if recursive and SHA256.